goodreads/templates/goodreads/explore_data.html

```diff
@@ -1,159 +1,22 @@
-                        initComplete: function () {
-                            // Hide the loading image once the DataTable is initialized
-                            $('#loadingImage').hide();
-                        }
-                    };
-                    // Initialize DataTable only after finishing modifications
-                    $('#rTable').DataTable(dataTableOptions);
-                },
-                error: function (xhr, errmsg, err) {
-                    // Handle any errors
-                    console.log(errmsg);
-                }
-            })
-        }
-    </script>
-
+{% load django_vite %}
 
+{% block content %}
     {% block dt %}
-        <script>
-            $(document).ready(function () {
-                load_datatable("{% url 'books-load-explore' %}",
-                    ["title_simple",
-                        "author",
-                        "gender",
-                        "nationality_chosen",
-                        "original_publication_year",
-                        "read",
-                        "narrative",
-                        "shelves",
-                        "number_of_pages"]);
-            });
-        </script>
-
-
-        <div class="headline">
+        <div class="headline mb-16">
             Explore Database
-            <br>
-        </div>
-        <img id="loadingImage" src="/static/novelty/N-Logo-loop.gif"
-             style="display: block; height: 30em; margin-left: auto; margin-right: auto;">
-        <style>
-            #rTable thead {
-                background-color: #f2f2f2;
-                color: #636efa;
-                font-weight: bold;
-                font-size: 0.8rem;
-            }
-        </style>
-        <div class="table-container">
-            <div class="table-wrapper" style="margin: 0;">
-                <table id="rTable" class="cell-border">
-                    <thead>
-                    <tr>
-                        <th>Title</th>
-                        <th>Author</th>
-                        <th>Gender</th>
-                        <th>Nationality</th>
-                        <th>Year</th>
-                        <th>Readers</th>
-                        <th>Narrative</th>
-                        <th>Shelves</th>
-                        <th># of Pages</th>
-                    </tr>
-                    </thead>
-                    <tbody>
-                    <!-- Table body content will be generated dynamically -->
-                    </tbody>
-                </table>
-                <br>
-                <br>
-            </div>
-            <br>
-            <br> <br>
-            <br>
         </div>
-        <br>
-        <br>
 
+        {% url 'books-load-explore' as dataUrl %}
+        <script id="dataTableProps" type="application/json">
+          {"url": "{{ dataUrl|escapejs }}"}
+        </script>
+
+        {% vite_asset "entryPoints/booksDataTable.tsx?elementId=dataTableProps" %}
```

Comment:
This is how an entry point gets loaded onto the page. In addition, the `elementId` URL param is used to pass some info to the module.

In this case, the element ID points to the `script` above, where we've encoded the props to pass into the React component. The entry point also renders the React component right after the `script` tag.